client/src/PublicRequest.ts

```diff
@@ -26,23 +26,34 @@ export type SingleKeyResult = {
 };
 
 export type TransactionInfo = {
-    sender: Uint8Array
-    senderType: Nimiq.Account.Type
-    recipient: Uint8Array
-    value: number
-    fee: number
-    validityStartHeight: number
-    recipientType?: Nimiq.Account.Type
-    data?: Uint8Array
+    keyPath: string,
+    senderLabel?: string,
+    sender: Uint8Array,
+    senderType: Nimiq.Account.Type,
+    recipient: Uint8Array,
+    recipientType?: Nimiq.Account.Type,
+    recipientLabel?: string,
+    value: number,
+    fee: number,
+    validityStartHeight: number,
+    data?: Uint8Array,
     flags?: number,
 };
 
+export enum BitcoinTransactionInputType {
+    DEFAULT = 'default',
+    HTLC_REDEEM = 'htlc-redeem',
+    HTLC_REFUND = 'htlc-refund',
+}
+
 export type BitcoinTransactionInput = {
     keyPath: string,
     transactionHash: string,
     outputIndex: number,
     outputScript: string,
     value: number,
+    witnessScript?: string,
+    type?: BitcoinTransactionInputType,
```

Comment:
`witnessScript` is the HTLC script necessary for HTLC redeem and refund transactions. `type` is used to tell the Keyguard which of the two it is, as the witness data the Keyguard has to generate is different with otherwise same request data.